packages/proxy/lib/http/util/service-worker-injector.ts

```diff
@@ -0,0 +1,228 @@
+/// <reference lib="dom" />
+
+import type { ServiceWorkerClientEvent } from './service-worker-manager'
+
+// this should be of type ServiceWorkerGlobalScope, but we can't reference it here
+// because this file is imported elsewhere and the ServiceWorker lib causes
+// errors in those other packages
+declare let self: typeof globalThis & {
+  __cypressServiceWorkerClientEvent: (event: string) => void
+  onfetch: (event: Event) => void
+}
+
+/**
+ * Injects code into the service worker to overwrite the fetch events to determine if the service worker handled the request.
+ * @param body the body of the service worker
+ * @returns the updated service worker
+ */
+export const injectIntoServiceWorker = (body: Buffer) => {
+  function __cypressInjectIntoServiceWorker () {
+    let _listenerCount = 0
```

Comment:
Why all the underscores here? This should be isolated from the rest of the script?